onnx/defs/image/defs.cc

```diff
@@ -0,0 +1,75 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <functional>
+#include "onnx/defs/data_type_utils.h"
+#include "onnx/defs/schema.h"
+#include "onnx/defs/tensor_proto_util.h"
+
+namespace ONNX_NAMESPACE {
+
+static const char* ImageDecoder_ver20_doc =
+    R"DOC(Loads and decodes and image from a file. If it can't decode for any reason (e.g. corrupted encoded stream, invalid format, it will return an empty matrix).
+
+The following image formats are supported: BMP, JPEG, JPEG2000, TIFF, PNG, WebP, Portable image format (PBM, PGM, PPM, PXM, PNM).
+
+Decoded images follow a channel-last layout: (Height, Width, Channels).
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
```

Comment:
I believe we discussed that earlier, and the conclusion was not to add a domain unless we have a significant number of operators that would justify it.

Reply:
Maybe a conclusion was reached somewhere else but to me this op feels very different than the other ones we use to create neural networks. It seems it would make sense mentally that a backend doesn't need to implement a decoding op for it to support core ONNX?

PyTorch and tensorflow both have image decoding operations scoped in a different namespace.

Reply:
I do agree with your points above. This operator seems special enough to justify a new "image" domain. I'll consider it and bring it for discussion with the preprocessing working group. Thanks.

Reply:
It is one operator now. What happens when the number is big enough to justify a new domain? Or maybe we could see that as an IO operator. We may have other similar operators to convert compress data into tensors, data being images, sound, video, ...

Reply:
I think one  important point is Justin's comment "a backend doesn't need to implement a decoding op for it to support core ONNX". More generally, what is the expectation/goal/intention of using other namespaces? Is there an implication that this op is not as important as the ones in the standard domain, and that core onnx backends may not implement it?

Reply:
What would be nice is some set of principles/guidelines for making the decision about which domain to place new ops in, something we haven't had to deal with so far (except for onnx vs onnxml, where there is some principle/guideline).

Reply:
Should it be generic like preprocessing or prepostprocessing? Or, specific to image?

Reply:
`ai.onnx.io` for anything related to images, sound, videos, ... ?

Reply:
It makes more sense to me to use `image` (and include image processing operations in the future) or `image_codec` (only image decoding/encoding) as domain. I can imagine a backend that wants to implement anything relate to image processing (including decoding) but has no means to implement audio/video operators. 

I agree that we need to decide and document our approach and implications of using domains.

Reply:
The decision for this PR is to leave it in the standard domain. We can consider adding ops to new domains once we have  (i) Infrastructure support (existing version-converter and function-op-definition mechanism both support the standard domain specially, but do not handle other domains), and (ii) There is a concrete proposal (about 1 or more new domains to be added, as well as the conventions to be used to decide when to put which op in which domain).